rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,259 @@
+
+#### Artifact Registry / Google Container Registry
+[GCR](https://cloud.google.com/container-registry) and [Artifact Registry](https://cloud.google.com/artifact-registry) are Google's container registry solution where the former is a subset of the latters capabilities. They support [vulnerability scanning](https://cloud.google.com/artifact-registry/docs/analysis) for a [cost](https://cloud.google.com/container-analysis/pricing), but lack tag immutability. 
+
+With support for multi-regional deployment, GCR / Artifact Registry provide a similar experience for both the Teleport team and our customers. However, due to Teleports stronger operational experience using AWS, we recommend ECR over GCR / Artifact Registry.
+
+### Appendix A: Example Terraform
+```hcl
+# Internal Repository for Teleport
+resource "aws_ecr_repository" "teleport" {
+  name                 = "gravitational/teleport"
+  image_tag_mutability = "IMMUTABLE"
+
+  encryption_configuration {
+    encryption_type = "KMS"
+    kms_key         = aws_kms_key.ecr_key.arn
+  }
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+# Public Repository for Teleport
+resource "aws_ecrpublic_repository" "teleport" {
+  repository_name = "teleport"
+
+   catalog_data {
+      ...
+   }
+}
+
+# Promotion User Policy
+data "aws_iam_policy_document" "teleport_promotion_user" {
+  statement {
+    sid    = "AllowPushImages"
+    effect = "Allow"
+    actions = [
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:CompleteLayerUpload",
+      "ecr:InitiateLayerUpload",
+      "ecr:PutImage",
+      "ecr:UploadLayerPart"
+    ]
+    resources = [aws_ecrpublic_repository.teleport.arn]
+  }
+  statement {
+    sid    = "AllowPullImages"
+    effect = "Allow"
+    actions = [
+      "ecr:BatchGetImage",
+      "ecr:GetDownloadUrlForLayer",
+    ]
+    resources = [aws_ecr_repository.teleport.arn]
+  }
+  statement {
+    sid    = "AllowAuthToken"
+    effect = "Allow"
+    actions = [
+      "ecr:GetAuthorizationToken",
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "teleport_promotion_user" {
+  name        = "teleport_promotion_user"
+  path        = "/"
+  description = "Amazon ECR Policy for promoting teleport from internal ECR to public ECR."
+  policy      = data.aws_iam_policy_document.teleport_promotion_user
+}
+
+resource "aws_iam_user" "teleport_promotion_user" {
+  name = "teleport_promotion_user"
+}
+
+resource "aws_iam_user_policy_attachment" "teleport_promotion_user" {
+  user       = aws_iam_user.teleport_promotion_user.name
+  policy_arn = aws_iam_policy.teleport_promotion_user.arn
+}
+```
+
+### Appendix B: Teleport Public Images
+The following table represents a best guess guide to migration of existing images from Quay to Harbor. They have been marked as such given their perceived relevance based on Quay activity and number of references in the Gravitational organization. 
+
+#### **Repositories to migrate and mirror**
```

Comment:
We are creating a Kubernetes Operator for Teleport
It'll have its own image: `teleport-operator`

I understand this document only reflects what we currently have, but can we add a note somewhere so we don't forget?
We expect to release the operator in the next couple of weeks